src/preprocess/links.rs

```diff
@@ -180,6 +227,70 @@ fn test_find_links_simple_link() {
                     }]);
 }
 
+#[test]
+fn test_find_links_with_range() {
+    let s = "Some random text with {{#include file.rs:10:20}}...";
+    let res = find_links(s).collect::<Vec<_>>();
+    println!("\nOUTPUT: {:?}\n", res);
+    assert_eq!(
+        res,
+        vec![
+            Link {
+                start_index: 22,
+                end_index: 48,
+                link: LinkType::IncludeRange(PathBuf::from("file.rs"), 10, 20),
+                link_text: "{{#include file.rs:10:20}}",
+            },
+        ]
+    );
+
+    let s = "Some random text with {{#include file.rs:10:}}...";
+    let res = find_links(s).collect::<Vec<_>>();
+    println!("\nOUTPUT: {:?}\n", res);
+    assert_eq!(
+        res,
+        vec![
+            Link {
+                start_index: 22,
+                end_index: 46,
+                link: LinkType::IncludeRangeFrom(PathBuf::from("file.rs"), 10),
+                link_text: "{{#include file.rs:10:}}",
+            },
+        ]
+    );
+
+    let s = "Some random text with {{#include file.rs::20}}...";
+    let res = find_links(s).collect::<Vec<_>>();
+    println!("\nOUTPUT: {:?}\n", res);
+    assert_eq!(
+        res,
+        vec![
+            Link {
+                start_index: 22,
+                end_index: 46,
+                link: LinkType::IncludeRange(PathBuf::from("file.rs"), 0, 20),
+                link_text: "{{#include file.rs::20}}",
+            },
+        ]
+    );
+
+    let s = "Some random text with {{#include file.rs::}}...";
```

Comment:
I'm probably going to mention this in the main discussion area as well, but I think it might be better to drop the `::` when you're wanting to include the entire file's contents. Then the "intuitive" way of writing this (`{{#include some_file.rs}}`) *Just Works*.

Reply:
There is no strong reason for this. I included this case because of the symmetry. The only (minor) reason, I have in mind, would be the case, when you are trying to find the right line numbers and are playing with them, ie. you are editing your include to see how the full file looks like, to go back to partial include after that.

Now, after describing the use case to you, I have a feeling, that it should be removed. :) What do you think?